flytekit/core/python_function_task.py

```diff
@@ -636,3 +642,116 @@ def run(self, remote: "FlyteRemote", ss: SerializationSettings, **kwargs):  # ty
+        return []
+
+
+eager_failure_task_resolver = EagerFailureTaskResolver()
+
+
+class EagerFailureHandlerTask(PythonAutoContainerTask, metaclass=FlyteTrackedABC):
+    _TASK_TYPE = "eager_failure_handler_task"
+
+    def __init__(self, name: str, inputs: typing.Optional[typing.Dict[str, typing.Type]] = None, **kwargs):
+        """ """
+        inputs = inputs or {}
+        super().__init__(
+            task_type=self._TASK_TYPE,
+            name=name,
+            interface=Interface(inputs=inputs, outputs=None),
+            task_config=None,
+            task_resolver=eager_failure_task_resolver,
+            secret_requests=[Secret(group="", key="EAGER_API_KEY")],  # todo: remove this before merging
+            **kwargs,
+        )
+
+    def dispatch_execute(self, ctx: FlyteContext, input_literal_map: LiteralMap) -> LiteralMap:
+        """
+        This task should only be called during remote execution. Because when rehydrating this task at execution
+        time, we don't have access to the python interface of the corresponding eager task/workflow, we don't
+        have the Python types to convert the input literal map, but nor do we need them.
+        This task is responsible only for ensuring that all executions are terminated.
+        """
+        # Recursive imports
+        from flytekit import current_context
+        from flytekit.configuration.plugin import get_plugin
+
+        most_recent = admin_common_models.Sort("created_at", admin_common_models.Sort.Direction.DESCENDING)
+        current_exec_id = current_context().execution_id
+        project = current_exec_id.project
+        domain = current_exec_id.domain
+        name = current_exec_id.name
+        logger.warning(f"Cleaning up potentially still running tasks for execution {name} in {project}/{domain}")
+        remote = get_plugin().get_remote(config=None, project=project, domain=domain)
+        key_filter = ValueIn("execution_tag.key", ["eager-exec"])
+        value_filter = ValueIn("execution_tag.value", [name])
+        phase_filter = ValueIn("phase", ["UNDEFINED", "QUEUED", "RUNNING"])
+        # This should be made more robust, currently lacking retries and exception handling
+        while True:
+            exec_models, _ = remote.client.list_executions_paginated(
+                project,
+                domain,
+                limit=100,
+                filters=[key_filter, value_filter, phase_filter],
+                sort_by=most_recent,
+            )
+            logger.warning(f"Found {len(exec_models)} executions this round for termination")
+            if not exec_models:
+                break
+            logger.warning(exec_models)
+            for exec_model in exec_models:
+                logger.warning(f"Terminating execution {exec_model.id}, phase {exec_model.closure.phase}")
+                remote.client.terminate_execution(exec_model.id, f"clean up by parent eager execution {name}")
+            time.sleep(0.5)
```

Comment:
should this be configurable?

Reply:
Can we make it configurable the first time someone asks to change it?  Let me make it a constant for now.  I just feel like configuration for everything can get overwhelming, esp. when we're not even sure if this whole feature will be removed.